logbook-spring-boot-starter/src/test/java/org/zalando/logbook/spring/ExcludeTest.java

```diff
@@ -0,0 +1,97 @@
+package org.zalando.logbook.spring;
+
+/*
+ * #%L
+ * Logbook: Spring
+ * %%
+ * Copyright (C) 2015 Zalando SE
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.zalando.logbook.Correlator;
+import org.zalando.logbook.Logbook;
+import org.zalando.logbook.MockRawHttpRequest;
+import org.zalando.logbook.RawHttpRequest;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import static java.util.Optional.empty;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ContextConfiguration
+@TestPropertySource(properties = "spring.config.name = exclude")
+public final class ExcludeTest extends AbstractTest {
+
+    @Configuration
+    public static class TestConfiguration {
+
+        @Bean
+        public Logger httpLogger() {
+            final Logger logger = mock(Logger.class);
+            when(logger.isTraceEnabled()).thenReturn(true);
+            return logger;
+        }
+        
+        @Bean
+        public Predicate<RawHttpRequest> requestPredicate() {
+            return request -> 
+                    !"/health".equals(request.getRequestUri());
```

Comment:
> in a "real" application the request.getRequestUri() will contain the full URL, not only the path (i.e. http://localhost/health).

You're absolutely right.

> should the RawHttpRequest contain a getPath method to facility this predicate mapping?

Most implementations of that interface will only have the full URI at hand. In order to access just the path one could use `URI.create(..).getPath()`, but that will lead us back to issue #49:

> **Problem**
When broken or malicious clients issue HTTP requests with invalid request targets (e.g. wrongly escaped query parameters) in the HTTP Request Line, logbook fails by throwing a IllegalArgumentException/URISyntaxException. This results in unlogged HTTP requests and most probably in 500 Internal Server Error responses from HTTP backends.
>
> **Proposed fix**
Don't use java.net.URI as return type of BaseHttpRequest::getRequestUri, but a plain String. Implementation was straight forward, with one exception: Query Parameter Obfuscation. If the request URI is malformed, query parameters cannot be extracted in the usual way in order to apply the Obfuscator to them. Currently, malformed URIs are then forwarded as-is. This may not be desirable. What do you think?

I'm not sure what to do here.

Another, rather unrelated question: I introduced `RequestPredicates` in another pull request to the core. It offers some predicates to match/exclude requests. It has support to match URIs 100% (`String.equals`) and based on a pattern (`Pattern.matches`). Since the `AntPathMatcher` is a part of Spring I wouldn't want to include it in the core. What's the best course of action here? Introduce another module `logbook-matcher` that has a spring dependency just for that single class so users of the library can choose whether they want it or not?